rasa/core/events/__init__.py

```diff
@@ -1021,7 +1022,14 @@ def __eq__(self, other) -> bool:
         if not isinstance(other, ActionExecuted):
             return False
         else:
-            return self.action_name == other.action_name
+            # if both have e2e_text compare both action name and e2e text
+            if hasattr(self, "e2e_text") and hasattr(other, "e2e_text"):
```

Comment:
as said in the comment: It's ok to keep. I'd remove the `hasattr` calls though as they are not required and add tests for the method